sql1.txt

```diff
@@ -32,12 +32,38 @@ SELECT CustomerName FROM Customers WHERE CustomerID NOT IN (SELECT CustomerID FR
 
 This is a subquery -- we haven't talked about those.  But, you can do the same thing with a left
 join by comparing the OrderID column with NULL.  Paste your SQL statement below.
-
+FIRST QUERY:
+ SELECT customername FROM customers c LEFT JOIN orders o ON c.customerid = o.customerid WHERE orderid IS NULL; 
+ Then I wanted to double check my answer with SECOND QUERY:
+ SELECT customername, orderid, c.customerid FROM customers c LEFT JOIN orders o ON c.customerid = o.customerid WHERE orderid IS NULL;
+ I found a discrepancy, 1 extra result:
+  ["Trail's Head Gourmet Provisioners ", nil, 82]]
+ I could not figure out why. I discovered it could be a trailing spaces issue, so I decided to use this:
+   SELECT TRIM(customername) FROM customers c LEFT JOIN orders o ON c.customerid = o.customerid WHERE orderid IS NULL;                                  
+[["TRIM(customername)"],                                                                                                                                     
+ ["Alfreds Futterkiste"],
+ ["Blauer See Delikatessen"],
+ ["Cactus Comidas para llevar"],
+ ["FISSA Fabrica Inter. Salchichas S.A."],
+ ["France restauration"],
+ ["Great Lakes Food Market"],
+ ["La corne d'abondance"],
+ ["Laughing Bacchus Wine Cellars"],
+ ["Lazy K Kountry Store"],
+ ["Let's Stop N Shop"],
+ ["Maison Dewey"],
+ ["North/South"],
+ ["Paris spécialités"],
+ ["Rancho grande"],
+ ["Spécialités du monde"],
+ ["The Cracker Box"],
+ ["Trail's Head Gourmet Provisioners"]]
+ I got them all!
```

Comment:
Oof what a frustrating problem you faced! The two queries you wrote are, for all intents and purposes, identical: they should select the same rows regardless of the columns from those rows your are asking to be returned. 

I wasn't able to replicate your discrepancy, however, I want to commend your problem solving on this! That kind of solution-oriented thinking is exactly the mindset we need as programmers! Way to go! 👏🏼

Reply:
Thank you.